x-pack/plugin/esql-core/src/test/java/org/elasticsearch/xpack/esql/core/optimizer/OptimizerRulesTests.java

```diff
@@ -369,1399 +106,4 @@ public void testFoldExcludingRangeWithDifferentTypesToFalse() {
-        FieldAttribute fa = TestUtils.getFieldAttribute("a");
-        FieldAttribute fb = TestUtils.getFieldAttribute("b");
-        FieldAttribute fc = TestUtils.getFieldAttribute("c");
-
-        Expression a1 = equalsOf(fa, ONE);
-        Expression a2 = equalsOf(fa, TWO);
-        And common = new And(EMPTY, equalsOf(fb, THREE), equalsOf(fc, FOUR));
-        And left = new And(EMPTY, a1, common);
-        And right = new And(EMPTY, a2, common);
-        Or or = new Or(EMPTY, left, right);
-
-        Expression exp = new BooleanSimplification().rule(or);
-        assertEquals(new And(EMPTY, common, new Or(EMPTY, a1, a2)), exp);
-    }
-
-    // (0 < a <= 1) OR (0 < a < 2) -> 0 < a < 2
-    public void testRangesOverlappingNoLowerBoundary() {
-        FieldAttribute fa = getFieldAttribute();
-
-        Range r2 = rangeOf(fa, L(0), false, TWO, false);
-        Range r1 = rangeOf(fa, L(0), false, ONE, true);
-
-        Or or = new Or(EMPTY, r1, r2);
-
-        CombineBinaryComparisons rule = new CombineBinaryComparisons();
-        Expression exp = rule.rule(or);
-        assertEquals(r2, exp);
-    }
-
-    public void testBinaryComparisonAndOutOfRangeNotEqualsDifferentFields() {
-        FieldAttribute doubleOne = fieldAttribute("double", DOUBLE);
-        FieldAttribute doubleTwo = fieldAttribute("double2", DOUBLE);
-        FieldAttribute intOne = fieldAttribute("int", INTEGER);
-        FieldAttribute datetimeOne = fieldAttribute("datetime", INTEGER);
-        FieldAttribute keywordOne = fieldAttribute("keyword", KEYWORD);
-        FieldAttribute keywordTwo = fieldAttribute("keyword2", KEYWORD);
-
-        List<And> testCases = asList(
-            // double > 10 AND integer != -10
-            new And(EMPTY, greaterThanOf(doubleOne, L(10)), notEqualsOf(intOne, L(-10))),
-            // keyword > '5' AND keyword2 != '48'
-            new And(EMPTY, greaterThanOf(keywordOne, L("5")), notEqualsOf(keywordTwo, L("48"))),
-            // keyword != '2021' AND datetime <= '2020-12-04T17:48:22.954240Z'
-            new And(EMPTY, notEqualsOf(keywordOne, L("2021")), lessThanOrEqualOf(datetimeOne, L("2020-12-04T17:48:22.954240Z"))),
-            // double > 10.1 AND double2 != -10.1
-            new And(EMPTY, greaterThanOf(doubleOne, L(10.1d)), notEqualsOf(doubleTwo, L(-10.1d)))
-        );
-
-        for (And and : testCases) {
-            CombineBinaryComparisons rule = new CombineBinaryComparisons();
-            Expression exp = rule.rule(and);
-            assertEquals("Rule should not have transformed [" + and.nodeString() + "]", and, exp);
-        }
-    }
-
-    // Equals & NullEquals
-
-    // 1 <= a < 10 AND a == 1 -> a == 1
-    public void testEliminateRangeByEqualsInInterval() {
-        FieldAttribute fa = getFieldAttribute();
-        Equals eq1 = equalsOf(fa, ONE);
-        Range r = rangeOf(fa, ONE, true, L(10), false);
-
-        PropagateEquals rule = new PropagateEquals();
```

Comment:
Any idea why here there are 27 removed tests that use `PropagateEquals` and only 24 in the esql `OptimizerRulesTests` class?

Reply:
I believe that's because there were three that tested `NullEquals`, which isn't implemented in ES|QL.  I'll double check that

Reply:
Confirmed.  The three tests that were in this file, but not migrated to `org.elasticsearch.xpack.esql.optimizer.OptimizerRulesTests` are:
- testEliminateRangeByNullEqualsInInterval
- testDualNullEqualsConjunction
- testEliminateRangeByNullEqualsOutsideInterval

All three of those test the behavior with `NullEquals`, which we don't support in ES|QL.  I think I deleted them when I was removing the stale `NullEquals` code a few months ago.